index.d.ts

```diff
@@ -0,0 +1,116 @@
+// Type definitions for core-decorators.js 0.10
+// Project: https://github.com/jayphelps/core-decorators.js
+// Definitions by: Qubo <https://github.com/tkqubo>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.1
+
+export interface ClassDecorator {
```

Comment:
Typescript comes with library file `lib.es5.d.ts`, which contains definitions similar to these.   Is there some reason to duplicate them here?

The typescript provided definitions read as follows.

```
declare type ClassDecorator = <TFunction extends Function>(target: TFunction) => TFunction | void;
declare type PropertyDecorator = (target: Object, propertyKey: string | symbol) => void;
declare type MethodDecorator = <T>(target: Object, propertyKey: string | symbol, descriptor: TypedPropertyDescriptor<T>) => TypedPropertyDescriptor<T> | void;
declare type ParameterDecorator = (target: Object, propertyKey: string | symbol, parameterIndex: number) => void;
```

Are these the definitions that will change due to changes made in advancing the proposal to stage 2?